share/ansible/roles/ci_run/tasks/alpine.yml

```diff
@@ -71,3 +71,10 @@
     src: /usr/local/src/shadow/etc/login.defs
     dest: /etc/login.defs
     remote_src: yes
+
+#The following is a workaround
+- name: Create a symbolic link
+  ansible.builtin.file:
+    src: /bin/bash
+    dest: /usr/bin/bash
```

Comment:
Ahh, so the problem is that they use bash in a path that doesn't exist in Alpine?  Good.  That's a smaller problem to deal with.  Adding a symlink sounds like a reasonable workaround.

LGTM.

Reply:
pytest-mh (CC: @pbrezina ) may want to use `#!/usr/bin/env bash` as their shebang, since that's significantly more portable.  It's technically not guaranteed by POSIX, but all systems I know have env(1) there.

Reply:
@ikerexxe , could you point at the line of code in pytest-mh that requires this workaround?  That would be very useful to note in the commit message.

Reply:
I suspect it's this one:
<https://github.com/next-actions/pytest-mh/blob/101d4d45a3f482d7de22d67edbfeb451cbedd2a4/pytest_mh/conn/__init__.py#L1040>

```py
        super().__init__("bash", "/usr/bin/bash -c")
```

and a few other similar ones.

@pbrezina , how about calling `/usr/bin/env bash -c` instead?

Reply:
As an easy example, my laptop is one of those systems where your command will break:

```sh
alx@devuan:~$ cat /etc/os-release 
PRETTY_NAME="Devuan GNU/Linux 5 (daedalus)"
NAME="Devuan GNU/Linux"
VERSION_ID="5"
VERSION="5 (daedalus)"
VERSION_CODENAME="daedalus"
ID=devuan
ID_LIKE=debian
HOME_URL="https://www.devuan.org/"
SUPPORT_URL="https://devuan.org/os/community"
BUG_REPORT_URL="https://bugs.devuan.org/"
alx@devuan:~$ which bash
/bin/bash
alx@devuan:~$ /usr/bin/bash -c "echo foo"
bash: /usr/bin/bash: No such file or directory
alx@devuan:~$ /usr/bin/env bash -c "echo foo"
foo
```

Reply:
Yes, please, open a ticket. I am also thinking about switching just to /bin since calls to other programs might break as well. All distros that done usrmerge have /bin as a symlink anyway.

Reply:
I think `/bin/bash` might be problematic in MacOS, where the version in /bin/bash is usually too old and problematic.  People usually install a more modern bash in /usr/local/bin/bash, or elsewhere.

And in the BSDs this might also be problematic, where they also install bash as a port in /usr/local/bin/bash, AFAIK.

I'll open a ticket.  :-)

Reply:
https://github.com/next-actions/pytest-mh/pull/96

Reply:
I removed the workaround as it was tackled in pytest-mh so I'm closing this conversation